core/math/math_funcs.h

```diff
@@ -199,6 +200,10 @@ class Math {
 		return value;
 	}
 	static _ALWAYS_INLINE_ int posmod(int p_x, int p_y) {
+		if (p_y == 0) {
+			ERR_FAIL_V_MSG(-1, "Cannot divide by zero");
+			return -1;
```

Comment:
`ERR_FAIL*` already returns, so the return is redundant (unreachable code).

I'm not sure `-1` is a good value for this though.

Reply:
It's only possible to return an integer value and I believe that most people define (or expect) `n (mod 0)` to be undefined, hence -1 since a negative number is impossible under normal circumstances.

Reply:
Maybe it should return `0`?